### PR TITLE
Update Judge upcoming_hearings query

### PR DESCRIPTION
**Pull request by joofsh**
This PR updates the SQL query we use for looking up all the hearings for a given judge. It uses the same `sdomainid` that Reader's welcome gate uses. 

connects #2590 

#### Test Plan:
- Verified local development with the "Hearing Prep" user works
- Manually run the query on UAT:

The executed SQL (confirming it's valid and includes sdomainid)
screen shot 2017-08-07 at 5 25 59 pm

Successfully returning hearings for the judge
screen shot 2017-08-07 at 5 26 17 pm

#### Note:
- Because we're including the migration in the same PR, this will temporarily break Hearing Prep on demo & UAT between the time the migration is run & when the new code is fully deployed. Should be max ~20min

CC @kierachell for awareness ^^

### CODE REVIEW

File: app/repositories/hearing_repository.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class HearingRepository
   class << self
     # :nocov:
-    def upcoming_hearings_for_judge(vacols_user_id, date_diff: 7.days)
```
Comment (joofsh) [inline app/repositories/hearing_repository.rb]: also removing the unnecessary `date_diff` parameter

File: app/models/vacols/case_hearing.rb [hunk 2]
```diff
@@ -53,11 +53,11 @@ class VACOLS::CaseHearing < VACOLS::Record
...
   # :nocov:
   class << self
-    def upcoming_for_judge(vacols_user_id, date_diff: 7.days)
-      id = connection.quote(vacols_user_id)
+    def upcoming_for_judge(css_id)
+      id = connection.quote(css_id)
 
       select_hearings
-        .where("staff.stafkey = #{id}")
+        .where("staff.sdomainid = #{id}")
```
Comment (joofsh) [inline app/models/vacols/case_hearing.rb]: this is the key change in this PR. This mirrors how Reader is doing the lookup

Comment (cmgiven) [inline app/models/vacols/case_hearing.rb]: 👍

File: lib/fakes/hearing_repository.rb [hunk 3]
```diff
@@ -4,9 +4,9 @@ class << self
     attr_accessor :hearing_records
   end
 
-  def self.upcoming_hearings_for_judge(vacols_user_id, date_diff: 7.days.ago)
-    user = User.find_by_vacols_id(vacols_user_id)
-    (hearing_records || []).select { |h| h.user_id == user.id && date_diff }
```
Comment (sharonwarner) [inline lib/fakes/hearing_repository.rb]: There's one more find_by_vacols_id on line 46.

**Pull request merged**